lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -991,6 +996,276 @@ def _generate_report(self, trans, invocation_id, **kwd):
             target_format=target_format,
         )
 
+    def _generate_invocation_bco(self, trans, invocation_id, **kwd):
+        decoded_workflow_invocation_id = self.decode_id(invocation_id)
+        workflow_invocation = self.workflow_manager.get_invocation(trans, decoded_workflow_invocation_id)
+        history = workflow_invocation.history
+        workflow = workflow_invocation.workflow
+        stored_workflow = workflow.stored_workflow
+
+        # pull in the user info from those who the history and workflow has been shared with
+        contributing_users = [stored_workflow.user]
+        for association in trans.sa_session.query(model.StoredWorkflowUserShareAssociation):
+            if association.user not in contributing_users:
```

Comment:
Agreed about consent & re-sharing concerns.

I see the purpose for sure, crediting users who contributed. The intention behind the code is good.

Including their name (which is often presented as "public" username) is one thing, but email isn't so acceptable. If the email is removed it's probably acceptable from a privacy standpoint, albeit potentially inaccurate due to your aforementioned concerns.

Reply:
I'm sorry for missing this - this isn't okay at all. I'm just going to drop everything but stored_workflow.user for now. @HadleyKing if you want to restore some of this - I think you're going to need to hide it behind a configuration flag or something.

I don't even think this is a great way to capture who contributed to the workflow - this is who the workflow is shared with. Once we are actually capturing author/license/etc.. information in the underlying models I think this will make a lot more sense to use.

Reply:
xref https://github.com/galaxyproject/galaxy/issues/10016

Reply:
Removed in https://github.com/galaxyproject/galaxy/pull/10217/files